include/openssl/rsa.h

```diff
@@ -192,15 +202,23 @@ int RSA_security_bits(const RSA *rsa);
 int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d);
 int RSA_set0_factors(RSA *r, BIGNUM *p, BIGNUM *q);
 int RSA_set0_crt_params(RSA *r,BIGNUM *dmp1, BIGNUM *dmq1, BIGNUM *iqmp);
+int RSA_set0_multi_prime_params(RSA *r, BIGNUM **primes, BIGNUM **exps,
+                                BIGNUM **coeffs, size_t pnum);
 void RSA_get0_key(const RSA *r,
                   const BIGNUM **n, const BIGNUM **e, const BIGNUM **d);
 void RSA_get0_factors(const RSA *r, const BIGNUM **p, const BIGNUM **q);
+size_t RSA_get_multi_prime_extra_count(const RSA *r);
+int RSA_get0_multi_prime_factors(const RSA *r, const BIGNUM **oth_primes);
 void RSA_get0_crt_params(const RSA *r,
                          const BIGNUM **dmp1, const BIGNUM **dmq1,
                          const BIGNUM **iqmp);
+int RSA_get0_multi_prime_crt_params(const RSA *r, const BIGNUM **oth_exps,
+                                     const BIGNUM **oth_coeffs);
 void RSA_clear_flags(RSA *r, int flags);
 int RSA_test_flags(const RSA *r, int flags);
 void RSA_set_flags(RSA *r, int flags);
+int32_t RSA_get_version(RSA *r);
+int RSA_set_version(RSA *r, int32_t ver);
```

Comment:
I'd like to argue in favour of minimizing of dependencies on headers. Even though int32_t is defined by standard it still takes an explicit #include. Yes, in this case it gets included, but that's not the point. Point is that if dependencies are minimal, then it would be much easier to clean up.

Reply:
I think these `int32_t` should be cleaned together with the `version` field of `struct rsa_st`.

Reply:
They don't have to. You can as well rely in implicit cast [which would be no-op anyway, because types are equivalent].